examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+  };
+
+  // An Image is an Element
+  this.Image = function(options) {
+    options = options || {};
+
+    var imageElement = new parent.Element(options);
+
+    imageElement._bitmap = rocky.gbitmap_create(options.url || '');
+    imageElement._bitmap.onload = function() { parent.mark_dirty(); };
+
+    imageElement._render = function(ctx, bounds) {
+      rocky.graphics_draw_bitmap_in_rect(ctx, this._bitmap, this.bounds);
+    };
+
+    return imageElement;
+  };
+
+  // A Window is an Element
+  this.Window = function(options) {
+    options = options || {};
+
+    var windowElement = new parent.Element(options);
+
+    windowElement._currentZ = 0;
+    windowElement._background = new parent.Rect(options);
+    windowElement._elements = [];
+
+    windowElement.show = function() {
+      var windowIndex = parent._windows.indexOf(this);
+        // Remove the window from the stack if it exists
+      if (windowIndex >= 0) {
+        parent._windows.splice(windowIndex, 1);
+      }
+
+      parent._windows.push(this);
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement.hide = function() {
+      var windowIndex = parent._windows.indexOf(this);
+      // Remove the window from the stack if it exists
+      if (windowIndex >= 0) {
+        parent._windows.splice(windowIndex, 1);
+      }
+
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement.add = function(el) {
+      var elementIndex = this._elements.indexOf(el);
+
+      // add a Z-Index if it's not already set
+      if (!el.z) el.z = this._currentZ++;
+
+      // Remove the elements from the stack if it exists
+      if (elementIndex >= 0) {
+        this._elements.splice(elementIndex, 1);
+      }
+
+      // Add the element to the list and sort
+      this._elements.push(el);
+      this._elements = this._elements.sort(function(a, b) {
+        return a.z - b.z;
+      });
+
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement.remove = function(el) {
+      var elementIndex = this._elements.indexOf(el);
+      // Remove the elements from the stack if it exists
+      if (elementIndex >= 0) {
+        this._elements.splice(elementIndex, 1);
+      }
+
+      parent.mark_dirty();
+      return this;
+    };
+
+    windowElement._defaultBackHandler = function(event) {
+      // Remove the current window
+      this.hide();
+    };
+
+    windowElement.onUp = function(cb) {
+      this._upHandler = cb;
+      return this;
+    };
+
+    windowElement.onDown = function(cb) {
+      this._downHandler = cb;
+      return this;
+    };
+
+    windowElement.onSelect = function(cb) {
+      this._selectHandler = cb;
+      return this;
+    };
+
+    windowElement.onBack = function(cb) {
+      if (cb) this._backHandler = cb;
+      else this._backHandler = this._defaultBackHandler;
+      return this;
+    };
+
+    windowElement._invokeButtonCallbackByCode = function(code, event) {
+      switch (code) {
```

Comment:
this can be simplified with an object
```
var handlers = {BACK_KEY: '_backHandler', …};
var func = handlers[code];
if (typeof func === 'function') { 
  func(event);
} else {
  console.log(…);
}
```